### DISCUSSION TITLE
unclear which parameters to pass into function updateHotspot()

### DISCUSSION THREAD

**Discussion by methot-ilya-3ds** [google/model-viewer#2138] [Q&A]
### Description

When attempting to change the HotSpot position using the updateHotspot() function passing in the following parameters do not seem to work. 

Example Code

```
var modelViewer = document.querySelector('#Model-Viewer');

modelViewer.updateHotspot(("hotspot-0", "(0, 0, 0)", "(0m -.1m 0m)"));
```

The function reaches this point in the code and always returns null:

```
updateHotspot(config) {
const hotspot = this[$hotspotMap].get(config.name);
            if (hotspot == null) {
                return;
            }
```

It seems that the function is looking for a class that has access to a .name property, but obviously passing in a string to the function as specified in the documentation would cause the function to always return null.

I have tried passing the function in other ways since the documentation is somewhat vague, but none of the options are able to return a value for config.name which isn't null.

#### Live Demo

https://volcano-legendary-crafter.glitch.me/

### Version

- model-viewer: v1.6.0

### Browser Affected

- [X] Chrome, version: xx.x.xxxx.xx
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [ ] Safari

### OS

- [ ] Android
- [ ] iOS
- [ ] Linux
- [ ] MacOS
- [X] Windows

**Comment by methot-ilya-3ds** [comment]
I have found a workaround by manually setting the position of the annotation. But I notice that the annotation's position does not re-render on screen even when changing the position in code.

```
var HotSpot = document.querySelector('#HotSpot');

HotSpot.dataset.position = String(x) + "," + String(y) + "," + String(z);
            
modelViewer.updateHotspot(HotSpot);
```

I figured that reaching the 

`this[$needsRender]();`

in the updateHotspot() method would cause the Hotspot to update, but this does not occur.

**Comment by elalish** [comment]
I think you just have a syntax error; you can see an example of using that method here: https://modelviewer.dev/examples/annotations/index.html#dimensions

You need to pass in an object with `name`, `position`, `normal` as properties.
